src/ControlSystem/ControlErrors/SizeControlInfo.hpp

```diff
@@ -0,0 +1,62 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+/// Labels the different 'state's of size control.
+///
+/// The different states are:
+/// - Initial: drives dr/dt of the excision boundary to
+///   SizeControlInfo::target_drift_velocity.
+/// - AhSpeed: drives the minimum characteristic speed on the excision boundary
+///   to SizeControlInfo::target_char_speed.
+/// - DeltaR: drives the minimum distance between the horizon and the excision
+///   boundary to be constant in time.
+enum class SizeControlLabel { Initial, AhSpeed, DeltaR };
```

Comment:
I don't know exactly how this will be used, but would it make sense to store the actual state object instead of a label?

Reply:
Storing the actual state object might work if we used a std::variant or a unique_ptr to the base class so that the things stored would all have the same type, but then each of the classes would need a virtual function that returns an ID telling what actual type it is.  Unless I'm missing something, that seems more complicated than an enum class.

Reply:
I was thinking a unique_ptr, yes.

It's hard for me to comment on what implementation makes sense when I don't know what you are going to do with this.  I don't know why  you want a type ID, for example.  Do you have something I can look at, even if it's still a work in progress?

Reply:
I think a unique_ptr would work.  I think the only time that I would need a type ID is for diagnostic output that would indicate what the current state is.

If a unique_ptr is stored in SizeControlInfo (which is serializable now), then presumably all the derived State classes need pup functions (which would be empty since the classes store no data)?

Reply:
Oh, I also use a type ID for the test (to check that the states are correct).  But without a type ID, I can check if `dynamic_cast` of the unique_ptr to a ptr of the correct type succeeds.

Reply:
You wouldn't need pup functions (those can be inherited), but you would need some macros and PUP_ID stuff.  Kind of annoying and maybe not worth the trouble.  I'll leave which way to go up to you.